### DISCUSSION TITLE
Multiple repositories in a codespace

### DISCUSSION THREAD

**Discussion by jirizavadil** [community/community#5159] [Codespaces] [answered]
Is there a way to add multiple repositories to a codespace?

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @jirizavadil, you can either enable a user level or org level setting to enable Codespaces have access to a single/multiple repos that you/your org owns. 
[Repository access for individual repos](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces)
[Repository access for org owned repos](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/managing-repository-access-for-your-organizations-codespaces)

Does this help? :)

**Reply by jirizavadil** [reply]
Hi, @tanmayeekamath , this looks like a good start, however i meant how you actually add multiple repos to the editor.

Use case: I have multiple repos using one another. On local machine each has its folder with git and they are symlinked together - all are JS projects linked by npm link. Idea is to replicate this in the web editor within one codespace.

Thanks!

**Reply by braem** [reply]
> Use case: I have multiple repos using one another. On local machine each has its folder with git and they are symlinked together - all are JS projects linked by npm link. Idea is to replicate this in the web editor within one codespace.

Instead of symlinks, perhaps they can be submoduled? Could get it into 1 repo then, even if you have to something like creating a separate 'root' repository with every repo submoduled.

**Reply by Chuxel** [reply]
You can place any repositories you have under `/workspaces`  and they will not be affected by rebuilding the container.  From there you can symlink as needed or use a [Multi-Root Worksapce](https://code.visualstudio.com/docs/editor/multi-root-workspaces#:~:text=%20Multi-root%20Workspaces%20%201%20Adding%20folders%20,when%20you%20are%20using%20multi-root%20workspaces%2C...%20More%20) instead.

**Reply by ryaninvents** [reply]
Should we expect authentication to work across multiple repos? I've got one repo open, and I'm trying to open the second one, but it's rejecting the SSH key.

**EDIT:** it works perfectly if you use `gh repo clone myorg/project`. Brilliant!

**Reply by Chuxel** [reply]
If you really need to have read/write access to many different repos, there is a workaround here. However, note that this does increase risk, so only use it with repositories you trust. Details: https://github.com/Chuxel/codespaces-multi-repo

**Reply by jirizavadil** [reply]
Thank you @Chuxel, the multi-repo is exactly what i needed!

**Reply by numbsafari** [reply]
Super helpful responses, @Chuxel.

**Reply by aaronsteers** [reply]
Does multi-root workspace work in `devcontainer.json`, or is there another way to implement this for codespaces?
